internal/providers/service.go

```diff
@@ -157,6 +165,123 @@ func (p *providerService) CreateGitHubOAuthProvider(ctx context.Context, provide
 	return &provider, nil
 }
 
+// CreateGitHubAppProvider creates a GitHub App provider with an installation ID
+func (p *providerService) CreateGitHubAppProvider(
+	ctx context.Context,
+	providerClass db.ProviderClass,
+	token oauth2.Token,
+	stateData db.GetProjectIDBySessionStateRow,
+	installationID int64,
+) (*db.Provider, error) {
+	installationOwner, err := p.getInstallationOwner(ctx, installationID)
+	if err != nil {
+		return nil, fmt.Errorf("error getting installation: %w", err)
+	}
+
+	tx, err := p.store.BeginTransaction()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error starting transaction: %v", err)
+	}
+	defer p.store.Rollback(tx)
+
+	qtx := p.store.GetQuerierWithTransaction(tx)
+
+	// Save the installation ID and create a provider
+	savedProvider, err := qtx.CreateProvider(ctx, db.CreateProviderParams{
+		Name:       fmt.Sprintf("%s-%s", providerClass, installationOwner.GetLogin()),
+		ProjectID:  stateData.ProjectID,
+		Class:      db.NullProviderClass{ProviderClass: providerClass, Valid: true},
+		Implements: app.Implements,
+		Definition: json.RawMessage(`{"github-app": {}}`),
+		AuthFlows:  app.AuthorizationFlows,
+	})
```

Comment:
nit: This `CreateProvider` looks like something we could have as a constructor specific for the class.

Reply:
Are you thinking we could add a `CreateProvider` to `github/app.go` and then the `Implements`, `AuthorizationFlows` and `Class` would be static? Then here we would call `app.CreateProvider(projectId, owner)`.